### PR TITLE
support forking existing chain

### PR DESCRIPTION
**Pull request by seunlanlege**

### CODE REVIEW

**Comment by seunlanlege** [comment]
cla:check

File: tests/polkadot.rs [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+   }};
+}
+
+pub struct CustomHostFunctions;
+
+impl sp_wasm_interface::HostFunctions for CustomHostFunctions {
+	fn host_functions() -> Vec<&'static dyn sp_wasm_interface::Function> {
+		use sp_wasm_interface::HostFunctions;
+		let mut host_functions = override_host_functions!(
+            "ext_crypto_ecdsa_verify_version_1", EcdsaVerify,
```
Comment (tomusdrw) [inline tests/polkadot.rs:69]: Formatting off here, please run `cargo fmt`.

**Review by tomusdrw — APPROVED** [review]
I lost track a bit with the project, but I trust it's providing value and working correctly. Glad that you have figured a way to fast-track the runtime upgrade - how long does it take now?

I think it would be cool to split the polkadot test code into separate crate and keep the runner more like a library.

**Comment by seunlanlege** [comment]
> how long does it take now

on a good day 15 minutes on the build host, 1hr 45 minutes on a bade day

**Pull request closed without merging**